### DISCUSSION TITLE
Such an awesome project

### DISCUSSION THREAD

**Discussion by meldarionqeusse** [bpatrik/pigallery2#328] [General]
Note sure were I can post this but I really need to express myself about how impressed with the performance of PiGallery2. I have been looking for a way to view my photos on the go without giving my photos to the service like google photos etc. I have an old slow NAS at home and I back up all my photos on it and I usually access it via smb from whatever device I want and it’s a painful awfully slow experience.

Over the past few years I have been trying to find something like this but I gave up and was concluding that I would have go tailor something myself. Then I found about about photonix and photoprism which could run on the raspberry pi but although great photo managers  they were slow to analyse all my photos by slow I mean slower than pigallery2 they are still faster than smb.

When I tried pigallery2 I had initially some problems but after opening an issue on github I was wuickly guided to a solution. Once I managed to pull the image and got it running I was blown away by the sheer speed. Crazy fast loading images. I mean even my ryzen 3000 pc sometimes does not thumbnails immediately in file explorer. And here I was on my phone connected via 3g over a vpn looking at ALL my photos each at least 30mb with pigallery2 running on raspberry pi 4 with an smb mount to my old slow NAS not local storage…. in seconds. All my photos organized how they make sense to me in the folder how I out them. Even when I go to a certain photo and zoom to maximum they load in less then 4 seconds over 3g network. To anyone slightly interested just try it out don’t hold back Very highly recommended.

I wish to thank all the developers of this great piece of software. Highly appreciate you work and effort. I  couldn’t find any were to send a donation to show my appreciation so I decided to write a sort of a review/thank you message.

Thank you

**Comment by bpatrik** [comment]
I'm happy to here that my hobby project so useful for others too. 

I never set up a donation button, because I always lazy to look into the taxing part. Also I'm making this app for my own entertainment, but I like to share it with others as the contributions and bug reports make the app better and also it does not cost anything to me :)

**Comment by dima-ser** [comment]
I would also like to express my gratitude and appreciation for this project, this is exactly what I needed. I love that it's read-only and allows me to use my own folder structure. And it supports existing XMP face tags! And even has the geolocation map! 

It allows me to use my existing photos with embedded metadata without locking me into an application or vendor that will be gone in a few years. And my data will continue to be usable outside of this application.

Thank you, thank you very much! Please keep it this way, don't try to make it something else.

**Comment by goooooouwa** [comment]
I absolutely love PiGallery2. Thank You @bpatrik for making such an awesome app!
